platform/security/idp/security-idp-plugin/security-idp-plugin-namequalifier/src/test/groovy/org/codice/ddf/security/idp/plugin/namequalifier/NameQualifierPresignPluginSpec.groovy

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.idp.plugin.namequalifier
+
+import ddf.security.samlp.SamlProtocol
+import org.apache.wss4j.common.saml.OpenSAMLUtil
+import org.opensaml.saml.saml2.core.Assertion
+import org.opensaml.saml.saml2.core.AuthnRequest
+import org.opensaml.saml.saml2.core.Issuer
+import org.opensaml.saml.saml2.core.Response
+import org.opensaml.saml.saml2.core.Subject
+import org.opensaml.saml.saml2.core.NameID
+import spock.lang.Specification
+
+
+class NameQualifierPresignPluginSpec extends Specification {
+
+    def ISSUER_ID = "https://localhost:8993/services/idp/login"
+
+    NameQualifierPresignPlugin plugin = new NameQualifierPresignPlugin()
+
+    // general mocks
+    AuthnRequest authNReq = Mock(AuthnRequest)
+    Issuer issuer = Mock(Issuer)
+
+    List<String> spMetadata = new ArrayList<>()
+    Set<SamlProtocol.Binding> bindings = new HashSet<>()
+
+    // set of mocks and concrete objects
+    Response response = Mock(Response)
+
+    List<Assertion> assertions = new ArrayList<>()
+    Assertion assertion = Mock(Assertion)
+
+    Subject subject = Mock(Subject)
+    NameID nameID = Mock(NameID)
+
+    def setupSpec() {
+        OpenSAMLUtil.initSamlEngine()
+    }
+
+    def setup() {
+
+        // set up authNReq
+        authNReq.getIssuer() >> issuer
+        issuer.getValue() >> ISSUER_ID
+
+        // set up name qualifier
+        response.getIssuer() >> issuer
+        response.getIssuer().setNameQualifier("TEST") >> true
+
+        // set up subject
+        assertion.getSubject() >> subject
+        assertion.getSubject().getNameID() >> nameID
+        assertion.getSubject().getNameID().setNameQualifier("TEST") >> true;
+
+        // set up basic response
+        response.assertions >> assertions
+        assertions.add(assertion)
+
+    }
+
+    def 'test name qualifier'() {
+
+        when:
+        plugin.processPresign(response, authNReq, spMetadata, bindings)
+
+        then:
+        1 * issuer.setNameQualifier('https://localhost:8993/services/idp/login')
+        1 * nameID.setNameQualifier('https://localhost:8993/services/idp/login')
```

Comment:
✏️ I think it's worth making the `NAME_QUALIFIER` field protected so that you can access it for this assertion. Instead of using a String literal.